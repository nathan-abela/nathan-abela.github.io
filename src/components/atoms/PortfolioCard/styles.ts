import styled from 'styled-components';

const StyledPortfolioCard = styled.article`
	margin-bottom: 48px;
	flex-direction: column;

	&:hover img {
		transform: scale(1.04);
	}

	a {
		color: ${({ theme }) => theme.headline};
		text-decoration: none;
	}

	span {
		border-radius: 4px;
	}

	img {
		transition: transform 0.3s ease;
	}

	h1 {
		margin: 0;
	}

	@media screen and (min-width: 768px) {
		display: flex;
		flex-direction: row-reverse;
		align-items: center;

		&:nth-of-type(odd) {
			flex-direction: row;

			.details {
				padding-left: 24px;
			}
		}

		&:nth-of-type(even) {
			.details {
				padding-right: 24px;
			}
		}

		& > * {
			width: 50%;
		}
	}
`;

export { StyledPortfolioCard };
