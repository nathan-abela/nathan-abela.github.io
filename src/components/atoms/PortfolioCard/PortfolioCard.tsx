import { FC } from 'react';
import Image from 'next/image';
import Link from 'next/link';

import { IPortfolio } from '@Types';
import { shimmer } from 'Utils/shimmer';
import { StyledPortfolioCard } from './styles';

export interface PortfolioProps {
	item: IPortfolio;
}

const PortfolioCard: FC<PortfolioProps> = ({ item }) => {
	return (
		<StyledPortfolioCard>
			<Link href={`/portfolio/${item.slug}`}>
				<a>
					<Image
						src={`/portfolio/${item.slug}.jpg`}
						width={1920}
						height={1080}
						objectFit="cover"
						alt={item.title}
						placeholder="blur"
						blurDataURL={shimmer(1920, 1080)}
					/>
				</a>
			</Link>

			<div className="details">
				<Link href={`/portfolio/${item.slug}`}>
					<a>
						<h1>{item.title}</h1>
						<p>{item.type}</p>
						<p>
							<strong>Technologies used:</strong> <em>{item.technologies.join(', ')}</em>
						</p>
					</a>
				</Link>
			</div>
		</StyledPortfolioCard>
	);
};

export { PortfolioCard };
