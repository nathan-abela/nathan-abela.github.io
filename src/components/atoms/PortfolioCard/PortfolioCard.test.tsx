import { render } from '@testing-library/react';

import { PortfolioCard } from './PortfolioCard';

describe('Portfolio Card', () => {
	it('renders correctly', () => {
		const { container } = render(
			<PortfolioCard
				item={{
					title: 'Sample Project',
					slug: 'sample-project',
					link: 'https://example.com',
					type: 'Web Development',
					technologies: ['HTML', 'CSS', 'JavaScript'],
					content:
						'This is a sample project description. Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
				}}
			></PortfolioCard>
		);
		expect(container).toMatchSnapshot();
	});
});
