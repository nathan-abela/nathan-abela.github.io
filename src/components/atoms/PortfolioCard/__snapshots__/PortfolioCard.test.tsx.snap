// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio Card renders correctly 1`] = `
<div>
  <article
    class="styles__StyledPortfolioCard-sc-1xdr3bq-0 dlrVCc"
  >
    <a
      href="/portfolio/sample-project"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%271920%27%20height=%271080%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Sample Project"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; filter: blur(20px); background-size: cover; background-image: url(data:image/svg+xml;base64,CiAgPHN2ZyB3aWR0aD0iMTkyMCIgaGVpZ2h0PSIxMDgwIiB2ZXJzaW9uPSIxLjEiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgeG1sbnM6eGxpbms9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkveGxpbmsiPgogICAgPGRlZnM+CiAgICAgIDxsaW5lYXJHcmFkaWVudCBpZD0iZyI+CiAgICAgICAgPHN0b3Agc3RvcC1jb2xvcj0icmdiYSgxMDAsIDEwMCwgMTAwLCAwLjIpIiBvZmZzZXQ9IjIwJSIgLz4KICAgICAgICA8c3RvcCBzdG9wLWNvbG9yPSJyZ2JhKDEwMCwgMTAwLCAxMDAsIDAuNSkiIG9mZnNldD0iNTAlIiAvPgogICAgICAgIDxzdG9wIHN0b3AtY29sb3I9InJnYmEoMTAwLCAxMDAsIDEwMCwgMC4yKSIgb2Zmc2V0PSI3MCUiIC8+CiAgICAgIDwvbGluZWFyR3JhZGllbnQ+CiAgICA8L2RlZnM+CiAgICA8cmVjdCB3aWR0aD0iMTkyMCIgaGVpZ2h0PSIxMDgwIiBmaWxsPSJyZ2JhKDEwMCwgMTAwLCAxMDAsIDAuMikiIC8+CiAgICA8cmVjdCBpZD0iciIgd2lkdGg9IjE5MjAiIGhlaWdodD0iMTA4MCIgZmlsbD0idXJsKCNnKSIgLz4KICAgIDxhbmltYXRlIHhsaW5rOmhyZWY9IiNyIiBhdHRyaWJ1dGVOYW1lPSJ4IiBmcm9tPSItMTkyMCIgdG89IjE5MjAiIGR1cj0iMXMiIHJlcGVhdENvdW50PSJpbmRlZmluaXRlIiAgLz4KICA8L3N2Zz4=); background-position: 0% 0%;"
        />
        <noscript />
      </span>
    </a>
    <div
      class="details"
    >
      <a
        href="/portfolio/sample-project"
      >
        <h1>
          Sample Project
        </h1>
        <p>
          Web Development
        </p>
        <p>
          <strong>
            Technologies used:
          </strong>
           
          <em>
            HTML, CSS, JavaScript
          </em>
        </p>
      </a>
    </div>
  </article>
</div>
`;
